addons/mail/static/tests/gif_picker/gif_picker_tests.js

```diff
@@ -226,3 +216,43 @@ QUnit.test("Searching for a GIF with a failling RPC should display an error", as
     await insertText("input[placeholder='Search for a GIF']", "search");
     await contains(".o-discuss-GifPicker-error");
 });
+
+QUnit.test(
+    "Scrolling at the bottom should trigger the search to load more gif, even after visiting the favorite.",
+    async (assert) => {
+        patchWithCleanup(GifPicker.prototype, {
+            get style() {
+                return "width: 200px;height: 200px;background: #000";
+            },
+        });
+
+        const pyEnv = await startServer();
+        const channelId = pyEnv["discuss.channel"].create({ name: "" });
+        const { openDiscuss } = await start({
+            mockRPC(route) {
+                if (route === "/discuss/gif/favorites") {
+                    assert.step("favorites");
+                }
+                if (route === "/discuss/gif/search") {
+                    assert.step("search");
+                    const _rpc = rpc.search;
+                    _rpc.results = gifFactory(4);
+                    return _rpc;
+                }
+                if (route === "/discuss/gif/categories") {
+                    assert.step("categories");
+                    return rpc.categories;
+                }
+            },
+        });
+        await openDiscuss(channelId);
+        await click("button[aria-label='GIFs']");
+        await click(".o-discuss-GifPicker div[aria-label='list-item']", { text: "Favorites" });
+        await click("i[aria-label='back']");
+        await click("img[data-src='https://media.tenor.com/6uIlQAHIkNoAAAAM/cry.gif']");
+        await contains(".o-discuss-Gif", { count: 4 });
+        await scroll(".o-discuss-GifPicker-content", "bottom");
+        await contains(".o-discuss-Gif", { count: 8 });
+        assert.verifySteps(["categories", "favorites", "search", "search"]);
```

Comment:
Better check presence of gif rather than `assert.step()`/`assert.verifySteps()`. This would make the test far more maintainable :)